cirq-google/cirq_google/api/v2/sweeps.py

```diff
@@ -171,6 +172,24 @@ def sweep_to_proto(
                 out.single_sweep.parameter.idx = sweep.metadata.idx
             if sweep.metadata and getattr(sweep.metadata, 'units', None):
                 out.single_sweep.parameter.units = sweep.metadata.units
+    elif isinstance(sweep, FiniteRandomVariable) and not isinstance(sweep.key, sympy.Expr):
+        sweep = cast(FiniteRandomVariable, sweep_transformer(sweep))
+        out.single_sweep.parameter_key = sweep.key
+        out.single_sweep.random_variable.length = sweep.length
+        out.single_sweep.random_variable.seed = sweep.seed
+        for random_value, prob in sweep.distribution.items():
+            out.single_sweep.random_variable.distribution[str(random_value)] = prob
+        # Encode the metadata if present
+        if isinstance(sweep.metadata, Metadata):
+            out.single_sweep.metadata.MergeFrom(metadata_to_proto(sweep.metadata))
+        else:
+            # Use duck-typing to support google-internal Parameter objects
+            if sweep.metadata and getattr(sweep.metadata, 'path', None):
+                out.single_sweep.parameter.path.extend(sweep.metadata.path)
+            if sweep.metadata and getattr(sweep.metadata, 'idx', None):
+                out.single_sweep.parameter.idx = sweep.metadata.idx
+            if sweep.metadata and getattr(sweep.metadata, 'units', None):
+                out.single_sweep.parameter.units = sweep.metadata.units
```

Comment:
This will make the 3rd place where this identical code appears in this function. Could we make things more DRY and factor out this code into a helper function? The helper function could be a subfunction inside `sweep_to_proto`, perhaps?

Reply:
Done.